src/pudl_archiver/archivers/eia/eiarecs.py

```diff
@@ -0,0 +1,231 @@
+"""Archive EIA Residential Energy Consumption Survey (RECS)."""
+
+import logging
+import re
+from dataclasses import dataclass
+from io import BytesIO
+from urllib.parse import urljoin
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+from pudl_archiver.frictionless import ZipLayout
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+
+@dataclass
+class LinkSet:
+    """Information a set of links in one tab of the RECS viewer.
+
+    See https://www.eia.gov/consumption/residential/data/2020/.
+    """
+
+    url: str
+    short_name: str
+    extension: str
+    pattern: re.Pattern
+    skip_if_html: bool = True
+
+
+def _url_for(year: int, view: str):
+    """Get the URL for a specific RECS year/tab combo."""
+    return (
+        f"https://www.eia.gov/consumption/residential/data/{year}/index.php?view={view}"
+    )
+
+
+# Each year, each tab's format changes. Rather than have complicated regexes that capture everything, just have lots of simple regexes
+YEAR_LINK_SETS = {
+    2020: {
+        "housing_characteristics": LinkSet(
+            url=_url_for(year=2020, view="characteristics"),
+            short_name="hc",
```

Comment:
I'd suggest using the view name (or some other something longer) instead of the short name in the file names so the files are more decipherable.

Reply:
yea if it were me i'd condense these two `LinkSet` attributes as just `view` and use the view name in the filename and construct the page url with `_url_for` in the `get_year_resources`